app/services/alma_availability_service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'rest-client'
+require 'nokogiri'
+
+class AlmaAvailabilityService
+  def initialize(mms_id)
+    @mms_id = mms_id
+    @response = query_availability
+    @xml = Nokogiri::XML(@response)
+  end
```

Comment:
is object state needed here? can they be passed in from outside, or otherwise eliminated?

Reply:
I guessed that we would be doing a lot of interaction with this API and that other methods would be added here. Thoughts to the contrary?

Reply:
the principal is to not have external dependencies in a constructor (i.e. don't do anything other than initialize state), further, try to make dependencies explicit (i.e. pass mms_id into methods that depend on it, rather than using internal state).  

if the @mms_id were not a state variable, and we didn't do anything interaction in the constructor, we can eliminate the constructor  all-together, and thusly all further methods can become static, and we have one less object on the heap to keep track of/gc

This would have the further advantage of making testing much more straightforward.

slack me for further details if you like.  this isn't a hard requirement, just a design principal I've found useful